### PR TITLE
Task - Update Applicant Model

### LINKED ISSUE #4841: Task - Update Applicant Model to include Classifications and other fields
# Description
We have a lookup table of Classifications, which are currently only attached to Job Posters. We want to allow Applicants to attach their current and past Classifications to their profile. This requires a new many-to-many relationship between Applicants and Classifications. 

Each Applicant-Classification needs a few more pieces of data: the level (an integer), and an integer representing the correct ordering. (We can assume the 'lowest' ordering represents the user's current classification.)

We also want to add veteran_status_id and citizenship_status_id fields to the Applicant, like what already exist on the JobApplication. These should be foreign keys to their respective lookup tables.

~Finally, also add a boolean is_gc_employee field.~

# Images
These changes are meant to support the new profile design:
https://xd.adobe.com/view/f66135de-6ab1-449d-ac7c-98ade3ce1fd2-a332/screen/22903d0a-6389-4e0b-a349-94d757852618

# Acceptance criteria
- [ ] Database is updated
- [ ] Model classes are updated

### PR DESCRIPTION
**Pull request by yonikid15**
Resolves #4841 

### Notes:
- Adds many-to-many relationship btw Applicant and Classification model.
- Adds two new fields to Applicant model: `citizenship_declaration_id` and `veteran_status_id`.
- I haven't added the `is_gc_employee` boolean field yet, since it hasn't been finalized yet (checkbox might be changed to radio button with 3 options). Also, we have a `not_in_gov` boolean field on the User model, which is set during registration.

### CODE REVIEW

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Good point that we're already storing GC Employee status in the User object, we shouldn't replicate that in Applicant. We might want to change how its stored in User though, even without a 3-option selection. 
Right now a user is only considered a GC employee if `gov_email` is not null, **and** `not_in_gov` is false. `not_in_gov` actually defaults to false for new Users, but that doesn't mean they _are in_ government, which is a bit confusing.

File: database/migrations/2020_12_03_032816_add_new_fields_to_applicants.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public function up()
+    {
+        Schema::table('applicants', function (Blueprint $table) {
+            $table->integer('citizenship_declaration_id')->unsigned()->nullable();
+			$table->integer('veteran_status_id')->unsigned()->nullable();
+            $table->foreign('citizenship_declaration_id')
+                ->references('id')
+                ->on('citizenship_declarations')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply database/migrations/2020_12_03_032816_add_new_fields_to_applicants.php:23]: Nice catch 😅

File: database/migrations/2020_12_03_032816_add_new_fields_to_applicants.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('applicants', function (Blueprint $table) {
+            $table->integer('citizenship_declaration_id')->unsigned()->nullable();
+			$table->integer('veteran_status_id')->unsigned()->nullable();
+            $table->foreign('citizenship_declaration_id')
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply database/migrations/2020_12_03_032816_add_new_fields_to_applicants.php]: Hmm, locally the indentation looks fine, but it's still off here.

File: app/Models/Applicant.php [hunk 3]
```diff
@@ -123,6 +130,13 @@ public function projects()
         return $this->morphMany(\App\Models\Project::class, 'projectable');
     }
 
+    public function classifications() //phpcs:ignore
+    {
+        return $this->belongsToMany(\App\Models\Classification::class, 'applicant_classification')
+            ->withPivot(['level', 'order'])
```
Comment (tristan-orourke) [inline app/Models/Applicant.php:136]: I personally find the Laravel withPivot functionality a bit confusing. I think I'd prefer to create a new model class which represent the pivot table along with the new data. We've already done this with other models like ExperienceSkills (which could have been a pivot table with `->withPivot(['justification'])`. 
I think having the separate model makes it easier to reason about the data relationship.
I'm open to having my mind changed on this idea, but that's my current opinion. What do you think?

Comment (yonikid15) [reply app/Models/Applicant.php]: I agree with you, thanks for clearing this up earlier!

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**